### PR TITLE
Fix conditional hook in hook builder false positive

### LINKED ISSUE #229: leancode_lint displays false warning for conditional hook for HookBuilder in if statement
image

### PR DESCRIPTION
**Pull request by shilangyu**
Closes #229

### CODE REVIEW

**Comment by shilangyu** [comment]
@KamilSztandur bump, I can make a release today

**Comment by shilangyu** [comment]
:(

File: packages/leancode_lint/test/lints_test_app/lib/avoid_conditional_hooks_test.dart [hunk 1]
```diff
@@ -211,3 +211,38 @@ final a = HookBuilder(
...
+}
+
+class IifeIsDetected extends HookWidget {
+  const IifeIsDetected({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    if (Random().nextBool()) {
+      return () {
+        // ignore: avoid_conditional_hooks
```
Comment (Albert221) [inline packages/leancode_lint/test/lints_test_app/lib/avoid_conditional_hooks_test.dart:240]: Did you mean `expect_lint`?

**Review by Albert221 — APPROVED** [review]

File: packages/leancode_lint/test/lints_test_app/lib/avoid_conditional_hooks_test.dart (hunk 1 above)
Comment (shilangyu) [reply packages/leancode_lint/test/lints_test_app/lib/avoid_conditional_hooks_test.dart:240]: 🤦 thank you

**Pull request merged**